draft-ietf-jsonpath-base.md

```diff
@@ -555,170 +559,29 @@ then the whole query selects no nodes.
-                      %x80-10FFFF       ; any non-ASCII Unicode character
-name-char = DIGIT / name-first
-
-DIGIT           =  %x30-39              ; 0-9
-ALPHA           =  %x41-5A / %x61-7A    ; A-Z / a-z
-~~~~
-
-Member names containing characters other than allowed by
-`dot-selector` — such as space (U+0020), minus (U+002D), or dot (U+002E)
-characters — MUST NOT be used with the `dot-selector`.
-(Such member names can be addressed by the
-`index-selector` instead.)
-
-#### Semantics
-{: unnumbered}
-
-The `dot-selector` selects the node of the member value corresponding
-to the member name from any JSON object in its input nodelist.
-
-It selects no nodes from any other JSON value.
-
-#### Examples
-{: unnumbered}
-
-JSON:
-
-    {"j": {"k": 3}}
-
-Queries:
-
-| Query | Result | Result Paths | Comment |
-| :---: | ------ | :----------: | ------- |
-| `$.j`   | `{"k": 3}` | `$['j']`      | Named value of an object      |
-| `$.j.k` | `3`        | `$['j']['k']` | Named value in nested object  |
-{: title="Dot selector examples"}
-
-### Dot Wildcard Selector {#wildcard}
+__PICKERS__ are defined solely within the context of the [child selector](#child-selector).
+A __PICKER__ outside of a child selector is invalid.
 
-#### Syntax
-{: unnumbered}
-
-The dot wildcard selector has the form `.*` as defined in the
-following syntax:
+A __PICKER__ identifies, or matches, child elements to be selected for the
+resulting nodelist.
 
-~~~~ abnf
-dot-wild-selector    = "." wildcard       ;  dot followed by asterisk
-wildcard             = "*"
-~~~~
-
-#### Semantics
-{: unnumbered}
+Each type of __PICKER__ functions differently and may be defined to operate on only
+objects, only arrays, or both.
 
-A `dot-wild-selector` acts as a wildcard by selecting the nodes of
-all member values of an object in its input nodelist as well as all
-element nodes of an array in its input nodelist.
+The relationship between a __PICKER__ and the child selector that contains it
+is further defined in the child selector section below.
 
-Applying the `dot-wild-selector` to a primitive JSON value (a number,
-a string, `true`, `false`, or `null`) selects no node.
+### Name __PICKER__ {#name-__PICKER__}
 
-#### Examples
-{: unnumbered}
-
-JSON:
-
-    {
-      "o": {"j": 1, "k": 2},
-      "a": [5, 3]
-    }
-
-Queries:
-
-| Query | Result | Result Paths | Comment |
-| :---: | ------ | :----------: | ------- |
-| `$.o.*` | `1` <br> `2` | `$['o']['j']` <br> `$['o']['k']` | Object values      |
-| `$.o.*` | `2` <br> `1` | `$['o']['k']` <br> `$['o']['j']` | Alternative result |
-| `$.a.*` | `5` <br> `3` | `$['a'][0]` <br> `$['a'][1]`     | Array members      |
-{: title="Dot wildcard selector examples"}
-
-### Index Selector
-
-#### Syntax {#syntax-index}
+#### Syntax {#syntax-name}
 {: unnumbered}
 
-An index selector `[<index>]` addresses at most one object member value or at most one array element value.
+A name __PICKER__ `'<name>'` matches at most one object member value.
```

Comment:
This section doesn't define what a name __PICKER__ is. I think it would be helpful to define a __PICKER__ in the ABNF and refer to it in the English.

Reply:
We already have a definition for a __PICKER__ in general.  This is the "name" variant.  The text explains what it does.  I'm not sure I understand what you're getting at.

Reply:
It was just an editorial comment. The text doesn't give any indication of the relationship between the name __PICKER__ and `quoted-member-name` (which is used in the following text and the ABNF without any reference to the name __PICKER__).

Reply:
What about the paragraph that immediately follows the text you commented on?

> Applying the `quoted-member-name` to an object value in its input nodelist, its string is required to match the corresponding member value. In contrast to JSON, the JSONPath syntax allows strings to be enclosed in _single_ or _double_ quotes.

Reply:
There just needs to be an explicit statement saying that `quoted-member-name` represents a name *picker. This needs to be reflected in the ABNF too.

Reply:
This was fixed, thanks.